### ISSUE TITLE
Expose the current render surface type

### ISSUE THREAD

**Issue by HakanL** [unoplatform/uno#13988]
### What would you like to be added

It would be great to have a property exposed that gets set to the current render surface type so a check can be made in the application if for some reason the environment has changed and software rendering is used, but not expected. I know it can be seen in the early log entries when set to trace, but for production it would be good with a check to make sure it doesn't revert to software rendering for some reason.

### Why is this needed

Catch environment errors early

### For which platform

Skia (GTK on Linux/macOS/Windows)

**Comment by HakanL** [comment]
I'm also aware that I can set the RenderSurfaceType on the host during startup, but this suggestion would be to just be able to quickly see the render type during testing/development.

**Comment by jeromelaban** [comment]
Thanks for the suggestion @HakanL! You should be able to tell using logging, by enabling the GtkHost log in the app filters. Did you try this already?

**Comment by HakanL** [comment]
Yes, I can see it in logging, but it could be a test in my code to make sure that other changes (docker file updates, dependencies, etc) don't break the render surface type where it would fall back to software, without enabling and sifting through logging. I could have a check in my release build to check the render type and error out if it's software. It would prevent an erroneous condition that may not get caught at the right time otherwise.

**Comment by mcNets** [comment]
@jeromelaban maybe I'm wrong, but `CompositionCapabilities.AreEffectsSupported()` should work. It's actually throwing a warning. (AreEffectsSupported is not implemented in Uno), but it returns true on a Windows machine. I haven't tested it on a Linux machine.

```csharp
CompositionCapabilities cc = new CompositionCapabilities();
var areEffectsSupported = cc.AreEffectsSupported();
```

It does not return the render surface type, but at least you can know if it's using software renderer.

https://github.com/unoplatform/uno/blob/e1aa19fc543adee4a4f2befb1665b11723152c54/src/Uno.UI.Composition/Composition/CompositionCapabilities.skia.cs#L12-L17

**Comment by HakanL** [comment]
Thanks @mcNets that at least tells me if it's accelerated, so that's good! However it would be nice to be able to get the type of renderer still, just so I can have safeguards in my code to make sure the application is running correctly after dependency updates, etc.

**Comment by mcNets** [comment]
@HakanL sorry I made a typo, it is `AreEffectsFast()` method.

**Comment by HakanL** [comment]
Yup, that's the one I used now, worked great (but my comment is still valid, to be able to determine in code what rendering type is used).

**Comment by mcNets** [comment]
`GtkHost` still publish `RenderSurfaceType`, the question is how to get `GtkHost` from `SkiaHostBuilder`.
```csharp
		/// <summary>
		/// Gets or sets the current Skia Render surface type.
		/// </summary>
		/// <remarks>If <c>null</c>, the host will try to determine the most compatible mode.</remarks>
		public RenderSurfaceType? RenderSurfaceType { get; set; }
````

**Comment by HakanL** [comment]
Yeah, short of using reflection I don't seem to have access to that.

**Comment by mcNets** [comment]
@HakanL can you check if this returns the surface type?
```csharp
    [STAThread]
    public static void Main(string[] args)
    {
        var host = SkiaHostBuilder.Create()
            .App(() => new App())
            .UseX11()
            .UseLinuxFrameBuffer()
            .UseMacOS()
            .UseWindows()
            .Build();

        var surface = ((WpfHost)host).RenderSurfaceType;

        host.Run();
    }
```

**Comment by HakanL** [comment]
@mcNets Hmm, in Windows (Skia) RenderSurfaceType is null, on my Linux (RPi) it's `Uno.WinUI.Runtime.Skia.X11.X11ApplicationHost` and it doesn't have a RenderSurfaceType exposed. Also while debugging I get this warning, is that accurate? I have migrated to the latest SDK/single-project structure. `If you are seeing this, make sure to follow the "Migrating WpfHost" section of Migrating from previous releases article in the Uno Platform documentation at https://aka.platform.uno/uno5-wpfhost-migration. WpfHost is used at the application level instead of window level starting Uno Platform 5.0.`

**Comment by jeromelaban** [comment]
@HakanL do you need to change the render surface type? As for the message on migration, you may want to migrate to an uno.sdk project structure, it's easier to manage.

**Comment by HakanL** [comment]
@jeromelaban Not necessarily, but it would be a nice feature to perhaps have a config option when calling `.UseX11()` where the surface could be changed. No, at this point I just wanted to be able to read what the current render type is so I can have a check to make sure it doesn't accidentally get set to software when deployed (due to incorrect dependencies or what not).
I have migrated to Uno.SDK (on 5.4 now) structure with a single project and using the desktop code:
```
            var host = SkiaHostBuilder.Create()
                .App(() => new App())
                .UseX11()
                .UseWindows()
                .Build();
```
So I shouldn't see that warning then I assume?

**Comment by MartinZikmund** [comment]
Is this still relevant @HakanL ?

**Comment by HakanL** [comment]
> Is this still relevant [@HakanL](https://github.com/HakanL) ?

Hmm, is there a programmatic way to make sure hardware acceleration is used. Basically I'm looking for a way to add to my own internal health status data a warning if for some reason my app has reverted back to software rendering (without me catching it in testing). I realize that may not be that likely to just happen out of the blue, but let's say there are OS or dependency updates and I don't catch that then it could happen, and I'd just like to know for sure that software rendering is not used. If `AreEffectsFast()` is the way to do that then I'm satisfied, it just seems a little "risky" since I don't know what "fast" means, if I knew what renderer was used then I could feel confident.
